nativelib/src/main/scala/scala/scalanative/misc/CRegex.scala

```diff
@@ -0,0 +1,106 @@
+package scala.scalanative.misc
+package regex
```

Comment:
We've been struggling really hard not to introduce new APIs outside in core with the exception of those necessary for interop. The reason being is that as soon as you use any of the non-standard APIs your code becomes completely unportable wrt JVM and JS targets.